fontforge/sfd.c

```diff
@@ -1068,6 +1079,37 @@ static void SFDDumpImage(FILE *sfd,ImageList *img) {
     fprintf(sfd,"\nEndImage\n" );
 }
 
+#ifndef _NO_LIBPNG
+static void SFDDumpImagePNG(FILE *sfd,ImageList *img) {
+    GImage *image = img->image;
+    struct _GImage *base = image->list_len==0?image->u.image:image->u.images[0];
+    struct enc85 enc;
+    int i;
+    int imsize = (int) (base->image_type==it_true?3*base->width:base->bytes_per_line);
+    char* pngbuf;
+    size_t pnglen;
+
+    FILE *pngfp = open_memstream(&pngbuf, &pnglen);
+    GImageWrite_Png(image, pngfp, false);
+    fflush(pngfp);
```

Comment:
What should I do to get it working on OS X? Can I tell however we're compiling there to use glibc's version, or...?

Reply:
The easiest solution is probably to read and write from a tmp file on the model of `svg.c`. If you wanted to get more aggressive you could move `GImageWrite_Png` and `GImageRead_Png` over to glib GIO interfaces (they don't have many direct callers).

Reply:
Can I just vendorize [this](https://github.com/alexreg/libmemstream)?

Reply:
I definitely can't do what `svg.c` is doing. It will be too slow to write the PNG twice...

Reply:
actually that libmemstream isn't a bad approach for OS X for the time being. I recommend trying to get it to work (including some testing that will run on the OS X build) hacked in and then asking @jtanx how best to "package" it.

Reply:
You can do everything in memory, I was actually just playing around with this just then, before I realised this pr was up. You don't need anything fancy, just change how you use libpng. Give me an hour or so and I'll get something up.

Reply:
Clarifying, you can do everything in memory in a portable manner.

Reply:
OK. I don't have an OS X instance to test on so I don't know how to test my vendorization. Let's wait for @jtanx to reply, this is his wheelhouse.

Reply:
(D'oh, replied again on an old page. Hi @jtanx! What do you mean by &laquo;Give me an hour or so and I'll get something up.&raquo;? Will you be pushing on top of `ctrlcctrlv:PNG_in_SFD`?)

Reply:
I can, but I was thinking of just providing a diff or something

Reply:
Alright, I pushed my changes to https://github.com/jtanx/fontforge/commits/images.

Relevant commits are https://github.com/jtanx/fontforge/commit/bf2a7afec8bd167f2498377d1f809e56f63cace9 and https://github.com/jtanx/fontforge/commit/0d9741d32ad01a1c070c52508d710d7607d010d2

The image writing routine could be improved to avoid a malloc/memcpy but I was lazy, and I think it's good enough. FontForge has a growbuf, but that's in libfontforge, so you'd need to pull that into gutils, too much work.

The latter commit changes to use that, plus implement a few (but not all; I'll leave that to you) other changes.

Reply:
Thank you @jtanx! I'm all out of energy for the night, but will work on reconciling your commits with this PR tomorrow. At a glance it looks like I can get it working that way with minimal headaches